### PR TITLE
(QA-2578) Add rake task for testing package of pe-client-tools

### PR DESCRIPTION
**Pull request by zreichert**

### CODE REVIEW

File: acceptance/Rakefile [hunk 1]
```diff
@@ -70,3 +71,48 @@ task :acceptance do
   beaker += " #{opts}" if opts != ''
   sh beaker
 end
+
+namespace :ci do
+  namespace :test do
+
+    desc 'This task is used by CI to test the package pe-client-tools'
+    rototiller_task :package => [:set_test_target]do |t|
```
Comment (johnduarte) [inline acceptance/Rakefile]: This task does not trigger the `:host_config` task to execute `beaker-hostgenerator` after the target is set.
```
$ SHA=5a51d6b9d31c7ec2eb2af4ed6b196a4c205e20b0 SUITE_VERSION=1.0.4.81.g5a51d6b CLIENT_TEST_TARGET=windows2012r2-64 be rake ci:test:package

The CLI switch '--xml' will be used.
The CLI switch '--no-color' will be used.
The CLI switch '--debug' will be used.
The CLI flag '--hosts' will be used with value 'centos7-64.mdca-windows2012r2-64client.a'.
The CLI flag '--helper' will be used with value 'helper.rb'.
The CLI flag '--keyfile' will be used with value '/home/john.duarte/.ssh/id_rsa-acceptance'.
The CLI flag '--load-path' will be used with value 'lib'.
The CLI flag '--pre-suite' will be used with value 'setup/pe/pre_suite'.
The CLI flag '--tests' will be used with value 'tests/'.
The CLI flag '--preserve-hosts' will be used with value 'onfail'.
INFO: The environment variable: 'SHA' was found with value: '5a51d6b9d31c7ec2eb2af4ed6b196a4c205e20b0': The sha for pe-client-tools
INFO: The environment variable: 'SUITE_VERSION' was found with value: '1.0.4.81.g5a51d6b': The suite version used by Jenkins
INFO: The environment variable: 'pe_dist_dir' was found with value: 'http://neptune.puppetlabs.lan/2016.3/ci-ready': The location to download PE from example "http://neptune.puppetlabs.lan/20XX.X/ci-ready"

Hosts file 'centos7-64.mdca-windows2012r2-64client.a' does not exist.
Trying as beaker-hostgenerator input.
```

Comment (zreichert) [inline acceptance/Rakefile]: @johnduarte updated beaker version.  Please `bundle install` and try again.

File: acceptance/Gemfile [hunk 2]
```diff
@@ -11,8 +11,9 @@ def location_for(place, fake_version = nil)
 end
 
 gem "beaker-hostgenerator", *location_for(ENV['BEAKER_HOSTGENERATOR_VERSION'] || "~> 0.2")
-gem "beaker", *location_for(ENV['BEAKER_VERSION'] || '~> 2.8')
+gem "beaker", '~> 2.5'
```
Comment (johnduarte) [inline acceptance/Gemfile]: @zreichert how will the inclusion of `beaker-pe` be handled? Will the pending update https://github.com/puppetlabs/puppetdb-cli/pull/53 address this as you expect?

```
  if ENV["BEAKER_TYPE"] == 'pe'
    gem 'beaker-pe', '~> 0.9'
    gem 'scooter', '~> 3.2', :require => false
  end
```

File: acceptance/Rakefile (hunk 1 above)
Comment (johnduarte) [inline acceptance/Rakefile]: @zreichert fails in the same way.
```
Hosts file 'centos7-64.mdca-windows2012r2-64client.a' does not exist.
Trying as beaker-hostgenerator input

$ be beaker --version
      wWWWw
      |o o|
      | O |  2.50.0!
      |(")|
     / \X/ \
    |   V   |
    |   |   |
```

File: acceptance/Rakefile [hunk 3]
```diff
@@ -70,3 +71,48 @@ task :acceptance do
...
+      flags = [
+          {:name => '--xml', :is_boolean => true},
+          {:name => '--no-color', :is_boolean => true},
+          {:name => '--debug', :is_boolean => true},
+          {:name => '--hosts', :override_env => 'TEST_TARGET'},
+          {:name => '--helper', :default => 'helper.rb'},
+          {:name => '--keyfile', :default => "#{ENV['HOME']}/.ssh/id_rsa-acceptance", :override_env => 'BEAKER_KEYFILE'},
+          {:name => '--load-path', :default => 'lib', :override_env => 'BEAKER_LOADPATH'},
+          {:name => '--pre-suite', :default => 'setup/pe/pre_suite', :override_env => 'BEAKER_PRESUITE'},
+          {:name => '--tests', :default => 'tests/', :override_env => 'BEAKER_TESTSUITE'},
```
Comment (johnduarte) [inline acceptance/Rakefile]: `tests/` is not the proper default for pdb-cli. As written in https://github.com/puppetlabs/puppetdb-cli/pull/53 the default would key off of the `BEAKER_TYPE` variable and be "#{ENV['BEAKER_TYPE']}/tests/".
I am not sure if we want to track the `BEAKER_TYPE` value or just default to `pe/tests`.

File: acceptance/Gemfile (hunk 2 above)
Comment (zreichert) [inline acceptance/Gemfile]: @johnduarte Im not sure what the best practice is here.  Im not sure if the current pipelines emit the BAEKER_TYPE environment variable.

**Pull request closed without merging**